### PR TITLE
(PUP-8094) Puppet lookup ignores trusted fact rule for classifier

### PR DESCRIPTION
**Pull request by BobosilaVictor**
Previously, while puppet lookup was creating the node request, the classifier would collect all facts from a node, and then search for local trusted data with `Puppet.lookup(:trusted_information)`. The problem is that, whether we are using lookup in a local or remote context, the `trusted_information` would always be empty, and so, the classifier would gather trusted_data from the local node where the lookup command is ran.
(https://github.com/puppetlabs/classifier/blob/main/puppet/lib/puppet/indirector/node/classifier.rb#L62-L69)

This commit fixes this issue by using the trusted facts gathered from pdb (#8786), to override the `trusted_information` option, so that the block of code mentioned above doesn't use the local node as a base of collection for trusted data, unless we are using puppet lookup in a local context.

- [x] Write tests

### CODE REVIEW

File: lib/puppet/application/lookup.rb [hunk 1]
```diff
@@ -361,17 +361,22 @@ def generate_scope
...
-        ni.terminus_class = :plain
-        node = ni.find(node, facts: extra_facts)
-        ni.terminus_class = tc
-      end
+      temp_node = Puppet::Node.new(options[:node])
+      temp_node.parameters['clientcert'] = options[:node]
+      overrides = {
+          trusted_information: extra_facts.values['trusted']
+        }
+        Puppet.override(overrides) do
```
Comment (joshcooper) [inline lib/puppet/application/lookup.rb:369]: Ideally I think we want to construct a "full" trusted context in the same way that the compiler would do. Unfortunately, the compiler and CA may be different hosts, so we can't just look in the local filesystem for the  agent's cert. But we could query the CA for the cert:

```ruby
service = Puppet.runtime[:http].route_to(:ca)
cert = service.get_certificate(node)
trusted = Puppet::Context::TrustedInformation(true, node, cert)
Puppet.override(trusted_information: trusted) do
  ...
end
```

Then the classifier terminus will call [`to_h`](https://github.com/puppetlabs/classifier/blob/6e007c5760896c0beb6cf31cf6768e729f02370c/puppet/lib/puppet/indirector/node/classifier.rb#L74) on that.

One important thing is that if puppetserver is configured to use a source of trusted *external* facts, then those facts will also be added to the trusted info (and can be used for lookup classification) https://github.com/puppetlabs/puppet/blob/8b4331b9f5b4e548621b76796f082770de1778d9/lib/puppet/context/trusted_information.rb#L48

**Comment by BobosilaVictor** [comment]
jenkins please test this with servertests

**Comment by BobosilaVictor** [comment]
jenkins please test this with servertests

**Comment by BobosilaVictor** [comment]
jenkins please test this with servertests

File: acceptance/tests/parser_functions/puppet_lookup_cmd.rb [hunk 2]
```diff
@@ -2086,14 +2087,15 @@ def data()
   content => "[server]
 vardir = /opt/puppetlabs/server/data/puppetserver
 logdir = /var/log/puppetlabs/puppetserver
-rundir = /var/run/puppetlabs/puppetserver
+rundir = /var/run/puppetlabs/pupapetserver
```
Comment (GabrielNagy) [inline acceptance/tests/parser_functions/puppet_lookup_cmd.rb:2090]: what

**Comment by GabrielNagy** [comment]
jenkins please test this with servertests

**Review by GabrielNagy — APPROVED** [review]
LGTM :+1:

Tested with a PE deployment, without this PR trusted information was not accessible in the context of `puppet lookup`, therefore trusted facts could not be used for classification. With this PR, trusted information is correctly passed to the classifier.

**Comment by BobosilaVictor** [comment]
jenkins please test this with servertests

**Review by GabrielNagy — APPROVED** [review]

**Pull request merged**